src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -365,7 +365,7 @@ public Task AuthenticateAsClientAsync(SslClientAuthenticationOptions sslClientAu
 
             ValidateCreateContext(sslClientAuthenticationOptions, _userCertificateValidationCallback, _certSelectionDelegate);
 
-            return ProcessAuthentication(true, false, cancellationToken)!;
+            return ProcessAuthenticationAsync(true, false, cancellationToken);
```

Comment:
Nit: as long as you're editing the code, can you name the Boolean arguments?
```suggestion
            return ProcessAuthenticationAsync(isAsync: true, isApm: false, cancellationToken);
```